actions/apt_action.go

```diff
@@ -49,6 +56,49 @@ func (apt *AptAction) Run(context *debos.DebosContext) error {
 	aptOptions = append(aptOptions, "install")
 	aptOptions = append(aptOptions, apt.Packages...)
 
+	if apt.LocalPackagesDir != "" {
+		localpackagesdir := path.Join(context.RecipeDir, apt.LocalPackagesDir)
+		destination, err := debos.RestrictedPath(context.Rootdir, "/usr/local/debs")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(destination)
+
+		err = debos.CopyTree(localpackagesdir, destination)
+		if err != nil {
+			return err
+		}
+
+		currentDir, _ := os.Getwd()
+		os.Chdir(destination)
+
+		// apt-ftparchive tries to read /etc/apt/apt.conf.d, add this path temporarily to prevent warnings
+		os.MkdirAll("/etc/apt/apt.conf.d", 0755)
+		defer os.RemoveAll("/etc/apt/apt.conf.d/")
+
+		err = debos.Command{}.Run("apt", "sh", "-c", "apt-ftparchive packages . > Packages")
+		if err != nil {
+			return err
+		}
+		err = debos.Command{}.Run("apt", "sh", "-c", "apt-ftparchive release . > Release")
+		if err != nil {
+			return err
+		}
+
+		os.Chdir(currentDir)
+
+		locallist, err := debos.RestrictedPath(context.Rootdir, "/etc/apt/sources.list.d/local.list")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(locallist)
+		cmd := fmt.Sprintf("echo 'deb [trusted=yes] file:///usr/local/debs/ ./' > %s", locallist)
```

Comment:
again, subsequent `apt` actions would also have this configuration available, best to delete it after the action rather than after the recipe